### ISSUE TITLE
Add option for CSS autocompletion in <link> tags

### ISSUE THREAD

**Issue by FatehAK** [pucelle/vscode-css-navigation#9]
Is it possible to add auto completion for CSS in `<link>` tags. I currently use two extensions for auto completion [link](https://marketplace.visualstudio.com/items?itemName=gencer.html-slim-scss-css-class-completion&ssr=false#overview) and peeking [link](https://marketplace.visualstudio.com/items?itemName=pranaygp.vscode-css-peek) . It would be extremely handy if I could replace the two extensions with your extension alone if you could add support for `<link>` tags.

**Comment by pucelle** [comment]
Hi, the `CSS Navigation` currently provide auto completion for all the class and id selectors in your work space folder, Do you mean it support filter auto completion results to only in the `<link> tags in current document?

**Comment by pucelle** [comment]
Hi, FatehAK, Can you provide more information about:

 1. What your work spaces like, does the `<link>` tags reference sources that belong to another work space folder.
 2. Is there many selectors with duplicate names exist in different files? Such that you may get many useless selectors when not respect `<link>`?

I didn't find the importance of respecting `<link>` tags to filter definitions recently, So I need to know what your scenario is and what problem you have.

Thanks.

**Comment by FatehAK** [comment]
I just thought you could improve performance by respecting just `<link>` tags and imports rather than scanning the entire workspace (which may auto complete unused CSS files as well).
On a side note does your extension support auto completion for remote style sheets?

**Comment by pucelle** [comment]
Hi, FatehAK, thanks for your reply.

Performance is what I firstly considered, That's one of the reason I didn't choose `CSS Peek` but write a new one. (Another reason is it doesn't support Sass & Less). I did benchmark on the plugin, result is here: [Stress Test & Performance](https://github.com/pucelle/vscode-css-navigation#stress-test--performance), And I also noted that [Why I started this project](https://github.com/pucelle/vscode-css-navigation#why-started-this-project).

I believe for normal usage (e.g., total CSS file size less than 10M), performance is not a problem. The true challenge is the back end architecture, which I already [marked it here].(https://github.com/pucelle/vscode-css-navigation#many-duplicate-definotions-got-can-you-limit-the-results-to-only-in-files-specified-by-link-tags-or-check-parent-nodes-to-ensure-the-whole-selector-match), This is main reason I didn't respect  `<link>` tags.

So if I really want to implement this, I may add a new option described as `Find CSS definitions only in "<link>" tags for HTML documents`. If you have some better ideas, it's welcome to mark them here.

Supports remote style sheets should be a good idea. I will support it later.
